lib/foreman_maintain/concerns/upstream.rb

```diff
@@ -0,0 +1,109 @@
+module ForemanMaintain
+  module Concerns
+    module Upstream
+      # TO DO - have a special file to keep Foreman and Katello mapping
+      VERSION_MAPPING = {
+        '3.2' => '4.4',
+        '3.3' => '4.5',
+        'nightly' => 'nightly'
+      }.freeze
+
+      def server_url
+        if el?
+          'https://yum.theforeman.org/'
+        elsif debian_or_ubuntu?
+          'https://deb.theforeman.org/'
+        else
+          raise 'Unknown operating system detected!'
+        end
+      end
+
+      def foreman_release_pkg_url(version)
+        if el?
+          "#{server_url}releases/#{version}/#{el_short_name}/x86_64/foreman-release.rpm"
+        elsif debian_or_ubuntu?
+          "#{server_url}pool/#{os_version_codename}/#{version}"\
+          '/f/foreman-release/foreman-release.deb'
+        else
+          raise 'Unknown operating system detected!'
+        end
+      end
+
+      def katello_pkgs_url(katello_version)
+        "#{server_url}katello/#{katello_version}/katello/#{el_short_name}/x86_64/"
+      end
+
+      def katello_release_pkg(version)
+        katello_version = VERSION_MAPPING[version]
+        "#{katello_pkgs_url(katello_version)}katello-repos-latest.rpm"
```

Comment:
I really wish we wouldn't need this mapping.
Can you please open a tracking redmine to "simplify" this? Probably by just placing the `katello-repos` RPM in both repos (foreman and katello)?

Reply:
yes, here we go: https://projects.theforeman.org/issues/35348 and [RFC](https://community.theforeman.org/t/introduction-of-a-new-katello-release-rpm/29749) you might have already checked that though

Reply:
I did not, wasn't following discourse the last few weeks ;)